src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -474,10 +474,14 @@ void PlayerMenu::SendQuestGiverQuestDetails(Quest const* quest, ObjectGuid npcGU
 void PlayerMenu::SendQuestQueryResponse(Quest const* quest) const
 {
     if (sWorld->getBoolConfig(CONFIG_CACHE_DATA_QUERIES))
-        _session->SendPacket(&quest->QueryData[static_cast<uint32>(_session->GetSessionDbLocaleIndex())]);
+    {
+        WorldPackets::Quest::QueryQuestInfoResponse* queryPacket = (WorldPackets::Quest::QueryQuestInfoResponse*)&quest->QueryData[static_cast<uint32>(_session->GetSessionDbLocaleIndex())];
+        queryPacket->Info.RewardMoney = _session->GetPlayer()->GetQuestMoneyReward(quest);
+        _session->SendPacket(queryPacket->Write());
```

Comment:
Very much thread unsafe and completely removes the point of this cache - its fine if we just send "as if at max level for quest" value here when cache is enabled